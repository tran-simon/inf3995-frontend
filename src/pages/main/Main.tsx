import React, { useState } from 'react';
import { CFProvider } from '../../context/CFContext';
import Layout from './Layout';

function Main() {
  const [batteryLevel, setBatteryLevel] = useState();
  const [droneNumber, setDroneNumber] = useState(0);

  // useEffect(() => {
  //   const getStats = setInterval(() => {
  //     fetch(`${BACKEND_URL}/getStats`).then((res) => {
  //       if (res.ok) {
  //         res.json().then((data) => {
  //           const keys = Object.keys(data);
  //           setBatteryLevel(keys.length > 0 ? data[keys[0]] : undefined);
  //           setDroneNumber(keys.length);
  //         });
  //       }
  //     });
  //   }, 1000);
  //   return () => clearInterval(getStats);
  // });
  //
  // const startSimulation = () => {
  //   fetch(`${BACKEND_URL}/startSim`);
  // };
  //
  // const land = () => {
  //   fetch(`${BACKEND_URL}/land`);
  // };
  //
  // const takeOff = () => {
  //   fetch(`${BACKEND_URL}/takeOff`);
  // };

  return (
    <div>
      <CFProvider
        scan={() => {
          return new Promise((resolve) => setTimeout(resolve, 1000));
        }}
      >
        <Layout>
          test
          <h1>Crazyflie Control Center</h1>
          <p>Battery Level : {batteryLevel} %</p>
          <p>Number of drones: {droneNumber}</p>
          {/*<Button onClick={() => startSimulation()}>Start simulation</Button>*/}
          {/*<Button onClick={() => takeOff()}>Take Off</Button>*/}
          {/*<Button onClick={() => land()}>Land</Button>*/}
          {/*<Button*/}
          {/*  onClick={() => {*/}
          {/*    fetch(`${BACKEND_URL}/scan`);*/}
          {/*  }}*/}
          {/*>*/}
          {/*  Scan for Crazyflies*/}
          {/*</Button>*/}
          {/*<Button*/}
          {/*  onClick={() => {*/}
          {/*    fetch(`${BACKEND_URL}/ConnectSocket`).then((response) => {*/}
          {/*      return response.text();*/}
          {/*    });*/}
          {/*  }}*/}
          {/*>*/}
          {/*  Start Simulation*/}
          {/*</Button>*/}
        </Layout>
      </CFProvider>
    </div>
  );
}

export default Main;
