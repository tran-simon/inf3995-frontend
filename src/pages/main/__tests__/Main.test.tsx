import React from 'react';
import Main from '../Main';
import { create } from 'react-test-renderer';
import 'firebase/database';

jest.mock('firebase', () => ({
  database: jest.fn().mockReturnValue({
    ref: jest.fn().mockReturnValue({}),
  }),
}));

jest.mock('react-router-dom', () => ({
  useParams: () => ({
    mapId: 'MAP_ID',
  }),
  useHistory: jest.fn(),
}));

describe('Main', () => {
  it('can match snapshot', () => {
    expect(create(<Main />)).toMatchSnapshot();
  });
});
