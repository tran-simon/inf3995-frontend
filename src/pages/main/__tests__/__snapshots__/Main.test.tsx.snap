// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main can match snapshot 1`] = `
<div
  className="MuiBox-root MuiBox-root-8"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appBar-1 mui-fixed MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root makeStyles-drawerToolbarBtn-6 MuiIconButton-edgeStart"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
      </button>
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        Centre de contrôle des Crazyflies
      </p>
    </div>
  </header>
  <div
    className="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-2 makeStyles-drawerClose-4"
  >
    <div
      className="MuiPaper-root MuiDrawer-paper makeStyles-drawerClose-4 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      />
      <hr
        className="MuiDivider-root"
      />
      <div
        className="MuiBox-root MuiBox-root-9"
      >
        <ul
          className="MuiList-root MuiList-padding"
        />
      </div>
      <ul
        className="MuiList-root MuiList-padding"
      >
        <div
          aria-disabled={false}
          className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="MuiListItemIcon-root"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
              />
            </svg>
          </div>
          <div
            className="MuiListItemText-root"
          >
            <span
              className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              Balayer
            </span>
          </div>
        </div>
        <div
          aria-disabled={false}
          className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="MuiListItemIcon-root"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M17 6H7c-3.31 0-6 2.69-6 6s2.69 6 6 6h10c3.31 0 6-2.69 6-6s-2.69-6-6-6zm0 10H7c-2.21 0-4-1.79-4-4s1.79-4 4-4h10c2.21 0 4 1.79 4 4s-1.79 4-4 4zM7 9c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
              />
            </svg>
          </div>
          <div
            className="MuiListItemText-root"
          >
            <span
              className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              Rafraîchissement Automatique
            </span>
          </div>
        </div>
        <div
          aria-disabled={false}
          className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="MuiListItemIcon-root"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M17 6H7c-3.31 0-6 2.69-6 6s2.69 6 6 6h10c3.31 0 6-2.69 6-6s-2.69-6-6-6zm0 10H7c-2.21 0-4-1.79-4-4s1.79-4 4-4h10c2.21 0 4 1.79 4 4s-1.79 4-4 4zM7 9c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
              />
            </svg>
          </div>
          <div
            className="MuiListItemText-root"
          >
            <span
              className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              Mode Simulation
            </span>
          </div>
        </div>
        <hr
          className="MuiDivider-root"
        />
        <li
          className="MuiListItem-root MuiListItem-gutters"
          disabled={false}
        >
          <div
            className="MuiListItemIcon-root"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M17.56 14.24c.28-.69.44-1.45.44-2.24 0-3.31-2.69-6-6-6-.79 0-1.55.16-2.24.44l1.62 1.62c.2-.03.41-.06.62-.06 2.21 0 4 1.79 4 4 0 .21-.02.42-.05.63l1.61 1.61zM12 4c4.42 0 8 3.58 8 8 0 1.35-.35 2.62-.95 3.74l1.47 1.47C21.46 15.69 22 13.91 22 12c0-5.52-4.48-10-10-10-1.91 0-3.69.55-5.21 1.47l1.46 1.46C9.37 4.34 10.65 4 12 4zM3.27 2.5L2 3.77l2.1 2.1C2.79 7.57 2 9.69 2 12c0 3.7 2.01 6.92 4.99 8.65l1-1.73C5.61 17.53 4 14.96 4 12c0-1.76.57-3.38 1.53-4.69l1.43 1.44C6.36 9.68 6 10.8 6 12c0 2.22 1.21 4.15 3 5.19l1-1.74c-1.19-.7-2-1.97-2-3.45 0-.65.17-1.25.44-1.79l1.58 1.58L10 12c0 1.1.9 2 2 2l.21-.02.01.01 7.51 7.51L21 20.23 4.27 3.5l-1-1z"
              />
            </svg>
          </div>
          <div
            className="MuiFormControl-root MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink Mui-error Mui-error MuiFormLabel-filled"
              data-shrink={true}
            >
              URL du serveur
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline Mui-error Mui-error MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={true}
                autoFocus={false}
                className="MuiInputBase-input MuiInput-input"
                disabled={false}
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
                value="http://localhost:5000"
              />
            </div>
          </div>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="MuiBox-root MuiBox-root-10"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    />
    <main
      className="makeStyles-content-7"
    >
      <svg
        height="100%"
        id="map-viewport"
        onMouseDown={[Function]}
        onMouseMove={[Function]}
        onMouseUp={[Function]}
        onWheel={[Function]}
        style={
          Object {
            "backgroundColor": "grey",
            "position": "absolute",
          }
        }
        viewBox="0 0 100 100"
        width="100%"
      >
        <defs>
          <filter
            filterUnits="userSpaceOnUse"
            id="f1"
            x="0"
            y="0"
          >
            <feGaussianBlur
              in="SourceGraphic"
              stdDeviation="5"
            />
          </filter>
        </defs>
      </svg>
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-align-content-xs-space-between"
        style={
          Object {
            "height": "100%",
            "padding": "1rem",
          }
        }
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4"
          >
            <span
              className="MuiSlider-root MuiSlider-colorPrimary"
              id="slider-zoom"
              onMouseDown={[Function]}
            >
              <span
                className="MuiSlider-rail"
              />
              <span
                className="MuiSlider-track"
                style={
                  Object {
                    "left": "0%",
                    "width": "49.74874371859296%",
                  }
                }
              />
              <input
                type="hidden"
                value="100"
              />
              <span
                aria-orientation="horizontal"
                aria-valuemax={200}
                aria-valuemin={1}
                aria-valuenow={100}
                className="MuiSlider-thumb MuiSlider-thumbColorPrimary"
                data-index={0}
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseLeave={[Function]}
                onMouseOver={[Function]}
                role="slider"
                style={
                  Object {
                    "left": "49.74874371859296%",
                  }
                }
                tabIndex={0}
              />
            </span>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4"
          style={
            Object {
              "height": "50%",
            }
          }
        >
          <div
            className="MuiBox-root MuiBox-root-18 makeStyles-root-16"
          >
            <div
              className="MuiBox-root MuiBox-root-19 makeStyles-leftSection-17"
            >
              <span
                className="MuiSlider-root MuiSlider-colorPrimary MuiSlider-vertical"
                onMouseDown={[Function]}
              >
                <span
                  className="MuiSlider-rail"
                />
                <span
                  className="MuiSlider-track"
                  style={
                    Object {
                      "bottom": "0%",
                      "height": "50%",
                    }
                  }
                />
                <input
                  type="hidden"
                  value="0"
                />
                <span
                  aria-orientation="vertical"
                  aria-valuemax={100}
                  aria-valuemin={-100}
                  aria-valuenow={0}
                  className="MuiSlider-thumb MuiSlider-thumbColorPrimary"
                  data-index={0}
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseOver={[Function]}
                  role="slider"
                  style={
                    Object {
                      "bottom": "50%",
                    }
                  }
                  tabIndex={0}
                />
              </span>
              <button
                className="MuiButtonBase-root MuiIconButton-root"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiIconButton-label"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 8c-2.21 0-4 1.79-4 4s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm8.94 3c-.46-4.17-3.77-7.48-7.94-7.94V1h-2v2.06C6.83 3.52 3.52 6.83 3.06 11H1v2h2.06c.46 4.17 3.77 7.48 7.94 7.94V23h2v-2.06c4.17-.46 7.48-3.77 7.94-7.94H23v-2h-2.06zM12 19c-3.87 0-7-3.13-7-7s3.13-7 7-7 7 3.13 7 7-3.13 7-7 7z"
                    />
                  </svg>
                </span>
              </button>
            </div>
            <span
              className="MuiSlider-root MuiSlider-colorPrimary"
              onMouseDown={[Function]}
            >
              <span
                className="MuiSlider-rail"
              />
              <span
                className="MuiSlider-track"
                style={
                  Object {
                    "left": "0%",
                    "width": "50%",
                  }
                }
              />
              <input
                type="hidden"
                value="0"
              />
              <span
                aria-orientation="horizontal"
                aria-valuemax={100}
                aria-valuemin={-100}
                aria-valuenow={0}
                className="MuiSlider-thumb MuiSlider-thumbColorPrimary"
                data-index={0}
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseLeave={[Function]}
                onMouseOver={[Function]}
                role="slider"
                style={
                  Object {
                    "left": "50%",
                  }
                }
                tabIndex={0}
              />
            </span>
          </div>
        </div>
      </div>
    </main>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
      style={
        Object {
          "borderTop": "1px solid rgba(0, 0, 0, 0.12)",
          "margin": "auto",
          "marginBottom": 0,
          "position": "relative",
        }
      }
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <h2
          className="MuiTypography-root MuiTypography-h2"
        >
          Nombre de drones: 
          0
        </h2>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Décoller
          </span>
        </button>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Atterrir
          </span>
        </button>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Utiliser les faux crazyflies
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
